src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -151,36 +156,44 @@ private Task OnProjectFactoryCompletedAsync()
+                    // For non-cross targeting projects, we can use the current project context if the TargetFramework hasn't changed.
+                    // For cross-targeting projects, we need to verify that the current project context matches latest frameworks targeted by the project.
                     // If not, we create a new one and dispose the current one.
 
                     if (!_currentAggregateProjectContext.IsCrossTargeting)
                     {
-                        return _currentAggregateProjectContext;
-                    }
-
-                    var projectProperties = await _commonServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
-                    var targetFrameworks = (string)await projectProperties.TargetFrameworks.GetValueAsync().ConfigureAwait(false);
+                        newTargetFramework = (string)await projectProperties.TargetFramework.GetValueAsync().ConfigureAwait(false);
+                        if (StringComparers.PropertyValues.Equals(_currentTargetFramework, newTargetFramework))
+                        {
+                            return _currentAggregateProjectContext;
+                        }
 
-                    if (_currentAggregateProjectContext.HasMatchingTargetFrameworks(targetFrameworks))
-                    {
-                        return _currentAggregateProjectContext;
+                        // Dispose the old workspace project context for the previous target framework.
+                        await DisposeAggregateProjectContextAsync(_currentAggregateProjectContext).ConfigureAwait(false);
                     }
-
-                    // Check if the current project context is up-to-date for the current active and known project configurations.
-                    var activeProjectConfiguration = _commonServices.ActiveConfiguredProject.ProjectConfiguration;
-                    var knownProjectConfigurations = await _commonServices.Project.Services.ProjectConfigurationsService.GetKnownProjectConfigurationsAsync().ConfigureAwait(false);
-                    if (knownProjectConfigurations.All(c => c.IsCrossTargeting()) &&
-                        _currentAggregateProjectContext.HasMatchingTargetFrameworks(activeProjectConfiguration, knownProjectConfigurations))
+                    else
                     {
-                        return _currentAggregateProjectContext;
-                    }
+                        // Check if the current project context is up-to-date for the current active and known project configurations.
+                        var activeProjectConfiguration = _commonServices.ActiveConfiguredProject.ProjectConfiguration;
+                        var knownProjectConfigurations = await _commonServices.Project.Services.ProjectConfigurationsService.GetKnownProjectConfigurationsAsync().ConfigureAwait(false);
```

Comment:
It might change - there is no way to atomically query CPS for all known configurations and active project configuration. Additionally, the check here is merely a performance optimization, and if the subsequent HasMatchingTargetFrameworks check fails, we will just dispose and create the same workspace context again.